datafusion/sql/src/expr/unary_op.rs

```diff
@@ -45,7 +45,13 @@ impl<S: ContextProvider> SqlToRel<'_, S> {
                 {
                     Ok(operand)
                 } else {
-                    plan_err!("Unary operator '+' only supports numeric, interval and timestamp types")
+                    plan_err!("Unary operator '+' only supports numeric, interval and timestamp types").map_err(|e| {
+                        let spans = operand.spans();
+                        let span = if let Some(s) = spans.as_ref() { s.first() } else { None };
+                        let diagnostic =
+                        Diagnostic::new_error("Unary operator '+' only supports numeric, interval and timestamp types".to_string(), span);
+                        e.with_diagnostic(diagnostic)
```

Comment:
Could we add a note or help here, such as `Try using arrow_cast to convert the value to a numeric, interval, or timestamp type`?